internal/pkg/source/analyzer.go

```diff
@@ -37,17 +37,36 @@ var Analyzer = &analysis.Analyzer{
 	ResultType: reflect.TypeOf(new(ResultType)).Elem(),
 }
 
-func run(pass *analysis.Pass) (interface{}, error) {
-	ssaInput := pass.ResultOf[buildssa.Analyzer].(*buildssa.SSA)
-	taggedFields := pass.ResultOf[fieldtags.Analyzer].(fieldtags.ResultType)
-	fieldPropagators := pass.ResultOf[fieldpropagator.Analyzer].(fieldpropagator.ResultType)
+// TODO(do not merge): I would appreciate any suggestions for type / field names.
+// upstreamArgs bundles upstreamArgs results for cleaner function signatures.
+type upstreamArgs struct {
```

Comment:
> Consider this call: (which appears in a number of places) [...]
> I think ideally, the client code would look like this: `classifier.IsSourcetype(t)`

This is what I was trying to touch upon in #240.  We should be more conscientious about what our exported interface is, and optionally, we may wish to bundle that interface as the `ResultType`'s exported methods to reduce signatures.  #240 was trying to get there in an iterative pace, but I could've done better in drawing attention to the end target.  Summarizing what you said there, I agree that it feels like that `propagation` is too familiar with the internals of how we're identifying sources here.  Something like `classifier.IsSourceType(t)` is a good target for iteration.

That said, I see `classifier` being the result type from the `source` analyzer so we can encapsulate the various conf / tags / current and future args rather than rely on tucking globals away.

------

> I also feel like this struct shouldn't include the built SSA. There is a significant mismatch in abstraction level between that field and the other fields: ssa is raw low-level material, while tags, props and conf are fairly high-level abstractions provided by our own code.

\shrug.  I personally thought that `conf` seemed like the odd one out in this bundle.  The others are all results from upstream analyzers, but `conf` is user input.  We could split one of those out if it makes for a better abstraction.  But if we were to split both of those out for a solid "Our upstream high level analyzer outputs" abstraction, I'm not sure shrinking signatures by one argument is worth the effort.

I'm not that bothered by signatures, though, and will defer to the opinion of y'all.